### PR TITLE
Fix few bugs and bump to v1.4.0

### PR DESCRIPTION
**Pull request by gabrielctn**
* Fix weblogo dependancy
* Fix unit test
* Migrate python2 to python3

### CODE REVIEW

**Comment by pierrepo** [comment]
Hi @gabrielctn :wave: 

Could you explain if this PR **preserve** or **breaks** back compatibility with Python 2?

All tests in CI failed because of
```
The command "sh -e /etc/init.d/xvfb start" failed and exited with 127 during .
```
It may worth we move to [GitHub actions](https://github.com/features/actions). @HubLot could you please help us with this?

**Comment by HubLot** [comment]
Hi,

Thanks for this PR ! It's great to have some improvements.

I agree with @pierrepo  we should move to actions! I gonna look into it. I prefer to not merge this until this is done.

I'm not a big fan of having merge commits from a fork into the original repo but it's not a big deal either. Maybe, for the next time, you can propose PR directly here. All of us will be happy to merge :)

If it breaks python 2 compatibility (which I don't mind), the new version should be 2.0.0 or at least 1.4.0.

**Comment by pierrepo** [comment]
Thanks @HubLot for your input.

I don't mind either if we break the Python 2 compatibility but we indeed must state it clearly (version 1.4.0 sounds good).
If we move forward, I'll add in the *Citation* section of the `README.md` that version 1.3.8 corresponds to the paper and will link to Zenodo and Sofware Heritage warehouses.

File: pbxplore/analysis/utils.py
Comment (jbarnoud) [inline pbxplore/analysis/utils.py:63]: Does this really need to be a list. I think it was used as an iterator when running on python 3.

File: pbxplore/analysis/visualization.py
Comment (jbarnoud) [inline pbxplore/analysis/visualization.py:18]: Does this change the version of weblogo that is required? If so, this should appear wherever we specify our dependencies (setup.py?)

File: pbxplore/analysis/visualization.py
Comment (jbarnoud) [inline pbxplore/analysis/visualization.py:165]: Does this need to be a list or is matplotlib happy with an iterator. I guess it was happy with the iterator otherwise it would not have worked on python 3 before.

File: pbxplore/structure/structure.py
Comment (jbarnoud) [inline pbxplore/structure/structure.py:99]: No need to make this a list explicitly. `dict` takes an iterable as its argument.

File: pbxplore/tests/__init__.py
Comment (jbarnoud) [inline pbxplore/tests/__init__.py:23]: Why the extra set of parentheses?

File: pbxplore/tests/__init__.py
Comment (jbarnoud) [inline pbxplore/tests/__init__.py:25]: Same as above: too many parentheses.

File: pbxplore/tests/__init__.py
Comment (jbarnoud) [inline pbxplore/tests/__init__.py:27]: Parentheses.

File: pbxplore/tests/__init__.py
Comment (jbarnoud) [inline pbxplore/tests/__init__.py:35]: Parentheses.

File: pbxplore/tests/__init__.py
Comment (jbarnoud) [inline pbxplore/tests/__init__.py:43]: Parentheses.

File: pbxplore/tests/test_functions.py
Comment (jbarnoud) [inline pbxplore/tests/test_functions.py:255]: What does mandate this change? The input did not change and the test used to pass. Did the behaviour of `pytest.approx` change? It would be reasonable.

**Review by jbarnoud — CHANGES_REQUESTED** [review]
Cool to see some life here! 

Contrary to @HubLot, I do not mind if the PR comes from a separate repo, using forks is the intended use of GitHub.

Some of the changes for python 3 seem unnecessary and automatically applied. Not all `range` need to be wrapped in `list`, and not all `print` need more parentheses.

File: pbxplore/analysis/visualization.py
Comment (HubLot) [reply pbxplore/analysis/visualization.py:18]: Yes, the change in the import name was made in Weblogo 3.7. I don't think it's necessary to specify it in our dependencies since it's the last version on pypi.

**Comment by HubLot** [comment]
I updated the CI to GitHub Actions in PR #176.
Besides the issue with Weblogo import name, it's working but with the lost of Python 2.7 and 3.4

So, I suggest to first merge this PR (once question raised solved) and after mine.

File: pbxplore/analysis/visualization.py
Comment (jbarnoud) [reply pbxplore/analysis/visualization.py:18]: It can be a problem for users who perform an upgrade rather than install from scratch. In that case, they may have PBxplore upgrading, but weblogo remaining at the previous version: then they would end up with a broken PBxplore and an unhelpful error message ("Cannot import weblogo" while the weblogo package is installed).

File: pbxplore/tests/test_functions.py
Comment (HubLot) [reply pbxplore/tests/test_functions.py:255]: I confirm this test still passed with Python >=3.6 without this modification.

File: pbxplore/analysis/visualization.py
Comment (HubLot) [reply pbxplore/analysis/visualization.py:18]: Good point.

File: pbxplore/analysis/utils.py
Comment (gabrielctn) [reply pbxplore/analysis/utils.py]: Indeed, 2to3 artifact, solved it

File: pbxplore/analysis/visualization.py
Comment (gabrielctn) [reply pbxplore/analysis/visualization.py:165]: Indeed, 2to3 artifact, solved it

File: pbxplore/tests/__init__.py
Comment (gabrielctn) [reply pbxplore/tests/__init__.py]: 2to3 script... I explained in a comment further

File: pbxplore/analysis/visualization.py
Comment (gabrielctn) [reply pbxplore/analysis/visualization.py:18]: Alright, then I will set the extra requirement as
```
# Extras requirements for optional dependencies
extras = {
    'analysis': ['weblogo>=3.7'],
    'all': ['weblogo>=3.7']
}
```

File: pbxplore/tests/test_functions.py
Comment (gabrielctn) [reply pbxplore/tests/test_functions.py:255]: I had issues with the unit test of this function, because `pytest.approx` was not accepting NaN comparisons, the behavior changed, you are still able to make it work @HubLot ?

File: pbxplore/analysis/visualization.py
Comment (HubLot) [reply pbxplore/analysis/visualization.py:18]: sounds good

File: pbxplore/tests/test_functions.py
Comment (HubLot) [reply pbxplore/tests/test_functions.py:255]: Yes, I was able to make it work on the version of the master with Python 3.6 to 3.8 with pytest 6.2.2
Maybe it was a bug in `pytest.approx` ?

**Comment by HubLot** [comment]
Thanks @gabrielctn  for the new commits. 

From what I see, here is what remains to be done:
  - Change bump version from 1.3.9  to 1.4.0
  - Update setup.py to precise the version of Weblogo
  - Update README to remove Python 3.4 support and add Python >=3.6
  - Update `docs/source/installation.rst with the above changes.

File: pbxplore/tests/test_functions.py
Comment (gabrielctn) [reply pbxplore/tests/test_functions.py:255]: Interesting, I run pytest 6.2.2 also, with python 3.8.5 and I got:
```
pbxplore/tests/test_functions.py::TestChainClass::test_get_phi_psi_angles[11-angles0] FAILED                                                [ 33%]
pbxplore/tests/test_functions.py::TestChainClass::test_get_phi_psi_angles[12-angles1] PASSED                                                [ 35%]
pbxplore/tests/test_functions.py::TestChainClass::test_get_phi_psi_angles[13-angles2] FAILED                                                [ 36%]

```

Comment (HubLot) [reply pbxplore/tests/test_functions.py:255]: Can you give the command and the first output lines of the pytest ?

Here's mine:

```bash
(PBxplore) [hub@Serenity:PBxplore (master=)]$ pytest
============================================================================================== test session starts ==============================================================================================
platform linux -- Python 3.8.5, pytest-6.2.2, py-1.10.0, pluggy-0.13.1 -- /home/hub/.local/share/virtualenvs/PBxplore-MAEcPXWb/bin/python
cachedir: .pytest_cache
rootdir: /home/hub/PBxplore, configfile: setup.cfg

[...]
pbxplore/tests/test_functions.py::TestChainClass::test_get_phi_psi_angles[11-angles0] PASSED                                                                                                              
pbxplore/tests/test_functions.py::TestChainClass::test_get_phi_psi_angles[12-angles1] PASSED                                                                                                              
pbxplore/tests/test_functions.py::TestChainClass::test_get_phi_psi_angles[13-angles2] PASSED
```

Comment (jbarnoud) [reply pbxplore/tests/test_functions.py:255]: Looks like there is a version dependency somewhere. The explicit form of the test is clearer and always works, let's go with that. I have the feeling the line is long, though; I would split it.
```suggestion
        assert (
            (angles["phi"] is None and phi_psi[resid]["phi"] is None)
            or angles["phi"] == pytest.approx(phi_psi[resid]["phi"])
        )
```

**Comment by jbarnoud** [comment]
It is not only the README that needs to be made aware of the python that are supported. If we want to play it modern, then we need to tell `setup.py` as well: https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires

File: pbxplore/tests/test_functions.py
Comment (gabrielctn) [reply pbxplore/tests/test_functions.py:255]: Ah yes that's it, for some reason even after upgrading all my pytest versions it was still using pytest-5.4.3 in the venv.
v6.2.2 works so it was indeed a bug in pytest version. Thanks for pointing it out @HubLot !
Still want to keep the new line or should I revert as proposes @jbarnoud ?

**Comment by gabrielctn** [comment]
Thank you for your reviewing !
All tests are passing on my side, if you see something missing please tell me, otherwise we're good to merge.

**Review by gabrielctn — COMMENTED** [review]
All changes were made accordingly

**Review by jbarnoud — APPROVED** [review]
Looks good to me.

**Comment by HubLot** [comment]
Good to me too!

**Review by HubLot — APPROVED** [review]

**Comment by pierrepo** [comment]
Looks very good to me too. Maybe I should merge #176 first.

**Comment by HubLot** [comment]
Readthedocs still has the previous version. I don't know if we need to trigger it or if it's done automatically. @pierrepo could you look into it ? Don't forger also to update the CITATION section :)

**Comment by pierrepo** [comment]
>Readthedocs still has the previous version. I don't know if we need to trigger it or if it's done automatically. @pierrepo could you look into it ? 

For some unknown raisons, I had to create the build of the doc manualy :astonished:  The doc is now up-to-date! 

> Don't forger also to update the CITATION section :)

On my way.

**Pull request merged**
